src/core_modules/capture-core/components/D2Form/FormBuilder/FormBuilder.component.js

```diff
@@ -390,7 +392,7 @@ export class FormBuilder extends React.Component<Props> {
         this.commitFieldUpdate({ fieldId, validators }, value, options);
     }
 
-    async commitFieldUpdate({ fieldId, validators, onIsEqual }: FieldCommitConfig, value: any, options?: ?FieldCommitOptions) {
+    async commitFieldUpdate({ fieldId, validators, onIsEqual }: FieldCommitConfig, value: any, options?: ?FieldCommitOptions, internalError?: ?Object) {
```

Comment:
We should spend a bit of time on some cleanup here to make this maintainable over time. Instead of adding a new param here in `internalError` we should use the `options` param we already have. We will have to add the `errorCode` to the `FieldCommitOptions`. Then we need to make some changes here: https://github.com/dhis2/capture-app/blob/c07e3015fddfa44544050a0e0393178aa8f0a010/src/core_modules/capture-core/components/D2Form/FormBuilder/FormBuilder.component.js#L446 and to the errorMessage two lines below. Currently, we run the validations, but if options.valid and options.error are set, we just override whatever the validations say. We should remove this override. This probably requires some investigation as we will have to figure out for what value types this logic is used and add a validator that does this for us. I.e., when we call validateField (https://github.com/dhis2/capture-app/blob/c07e3015fddfa44544050a0e0393178aa8f0a010/src/core_modules/capture-core/components/D2Form/FormBuilder/FormBuilder.component.js#L434) we should pass on the `options` object instead of `internalError`. For the value types that were using the override, we should add a validator that sets `valid` and `errorMessage` the normal way.

Reply:
Regarding the override of valid and error. Checked the git history and talked to Eirik, it is related to plugins and we should skip validations if we are committing from a plugin. I think the Formbuilder file should end up looking something like this: https://github.com/dhis2/capture-app/blob/DHIS2-15462-formbuilder/src/core_modules/capture-core/components/D2Form/FormBuilder/FormBuilder.component.js